lib/api/liskApi.js

```diff
@@ -163,22 +163,35 @@ LiskAPI.prototype.sendRequest = function (requestType, options, callback) {
 	return privateApi.sendRequestPromise.call(this, requestType, options).then(function (requestSuccess) {
 		var returnAnswer = (parseOfflineRequest(requestType, options).requestMethod === 'GET') ? requestSuccess.body : parseOfflineRequest(requestType, options).transactionOutputAfter(requestSuccess.body);
 
-		if (!callback || (typeof callback !== 'function')) {
-			return Promise.resolve(returnAnswer);
-		} else {
-			return callback(returnAnswer);
+		if (callback && (typeof callback === 'function')) {
+			callback(returnAnswer);
+		}
+		return returnAnswer;
+	}).then(function (result) {
+		if (!result.success && result.message.match(/Timestamp is in the future/) && !(options.timeOffset > 40e3)) {
+
+			var newOptions = {};
+			Object.keys(options)
+				.forEach(key => newOptions[key] = options[key]);
+			newOptions.timeOffset = (options.timeOffset || 0) + 10e3;
+			return new Promise(function (resolve, reject) {
+				setTimeout(function () {
+					that.sendRequest(requestType, newOptions, typeof callback === 'function' ? callback : null)
+						.then(resolve, reject);
+				}, 1000);
```

Comment:
Probably not. I put it in before limiting the number of requests that would be made but I've removed it now.